billing-tests/src/androidTest/java/com/mapbox/navigation/billing/test/BillingTests.kt

```diff
@@ -0,0 +1,137 @@
+import com.mapbox.navigation.testing.ui.utils.coroutines.startTripSessionAndWaitForActiveGuidanceState
+import com.mapbox.navigation.testing.ui.utils.getMapboxAccessTokenFromResources
+import com.mapbox.navigation.testing.ui.utils.runOnMainSync
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.take
+import kotlinx.coroutines.flow.toList
+import org.junit.After
+import org.junit.Assert
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import java.net.URI
+import java.util.concurrent.CancellationException
+import java.util.concurrent.TimeUnit
+
+/**
+ * Tests that are scheduled to run on CI. Needed to generate billing events
+ * that will later be checked during CI job to verify whether billing events appeared
+ * in the billing data base.
+ *
+ * In this tests we just generate billing events.
+ * More test cases and parameters verification can be found in 'TripSessionsBillingTest'.
+ */
+class BillingTests : BaseCoreNoCleanUpTest() {
+
+    @get:Rule
+    val mapboxNavigationRule = MapboxNavigationRule()
+
+    private lateinit var mapboxNavigation: MapboxNavigation
+
+    private lateinit var httpEventsObserver: HttpServiceEventsObserver
+
+    private val source = Point.fromLngLat(-77.031991, 38.894721)
+    private val destination = Point.fromLngLat(-77.030923, 38.895433)
+
+    override fun setupMockLocation(): Location = mockLocationUpdatesRule.generateLocationUpdate {
+        latitude = source.latitude()
+        longitude = source.longitude()
+        bearing = 190f
+    }
+
+    @Before
+    fun setup() {
+        httpEventsObserver = HttpServiceEventsObserver.register()
+
+        runOnMainSync {
+            assertTrue(httpEventsObserver.billingRequests().isEmpty())
+
+            mapboxNavigation = MapboxNavigationProvider.create(
+                NavigationOptions.Builder(context)
+                    .accessToken(getMapboxAccessTokenFromResources(context))
+                    .routingTilesOptions(
+                        RoutingTilesOptions.Builder()
+                            .tilesBaseUri(URI(mockWebServerRule.baseUrl))
+                            .build()
+                    )
+                    .build()
+            )
+        }
+    }
+
+    @After
+    fun cleanUp() {
+        HttpServiceEventsObserver.unregister()
+    }
+
+    @Test
+    fun test30SecondsLongActiveGuidanceTrip() = sdkTest(TimeUnit.MINUTES.toMillis(1)) {
+        val routes = requestRoutes(source, destination)
+
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(routes)
+        mapboxNavigation.startTripSessionAndWaitForActiveGuidanceState()
+
+        delay(TimeUnit.SECONDS.toMillis(30))
+
+        mapboxNavigation.clearNavigationRoutesAndWaitForUpdate()
+
+        waitForBillingHttpEvents(1) {
+            it.isActiveGuidanceSession
+        }
```

Comment:
Do we need this for e2e? I think, ideally, we shouldn't rely on implementation details at all in tests. I see our contract as:
```
start active guidance session
wait for 30 seconds
stop active guidance session
wait for a couple of seconds
```
what do you think about it? Does it make sense to you?

Reply:
Agreed with @VysotskiVadim to fix that and replace with 10 seconds delay.